plugins/pulp_rpm/plugins/importers/iso/importer.py

```diff
@@ -82,8 +82,13 @@ def import_units(self, source_repo, dest_repo, import_conduit, config, units=Non
 
         units = list(units)
 
+        # store the existing repo units to prevent querying mongo multiple times
+        repo_units = repo_controller.find_repo_content_units(dest_repo.repo_obj,
+                                                             yield_content_unit=True)
         for u in units:
-            import_conduit.associate_unit(u)
+            units_to_remove = [iso for iso in repo_units if iso['name'] == u['name']]
```

Comment:
I think this may be broken. repo_units is a generator; at the first iteration of this loop, won't that generator be exhausted, and then subsequent iterations will behave as though repo_units is empty?  Meaning that it only removes duplicates for the first unit in 'units'.

Tested it, attempting to iterate through a generator more than once indeed seems to give no results after the first run.

```
>>> def mygen():
...   for i in range(0, 10):
...     yield i
... 
>>> inst = mygen()
>>> print [x for x in inst]
[0, 1, 2, 3, 4, 5, 6, 7, 8, 9]
>>> print [x for x in inst]
[]
```

Reply:
Just to close out this thread as it was not automatically marked outdated by github - this was fixed by forcing repo_units to a list, a few lines above.